Add suffix to CsvEnsembleReader constructor

danhamill opened the pull request:
Working on the Feather-Yuba WCM update, we have been provided hindcast ensembles with different CSV filename patterns than what this library was developed with.  In this PR I have added `suffix` to the `CsvEnsembleReader` constructor and updated all tests.  This will allow people to specify different filename suffixes when importing ensembled time series to a database.

MikeNeilson reviewed (commented):
It would be good to include some of the new data, using the new suffix so others are aware of what it is and data is available for any additional tests that may be required, or within existing tests in cases of subtle differences.

danhamill commented:
> It would be good to include some of the new data, using the new suffix so others are aware of what it is and data is available for any additional tests that may be required, or within existing tests in cases of subtle differences.

What do you think about building in watershed specific subdirectories in `FIRO_TSEnsembles/src/test/resources/hefs_cache`

It appears `hec.ensemble.Testing#bulkTesting` has been set up to look at multiple watersheds, but there are no `FeatherYuba` or `EastSierra` hefs csv files in the resources.

MikeNeilson commented:
> What do you think about building in watershed specific subdirectories in `FIRO_TSEnsembles/src/test/resources/hefs_cache`
> 
> It appears `hec.ensemble.Testing#bulkTesting` has been set up to look at multiple watersheds, but there are no `FeatherYuba` or `EastSierra` hefs csv files in the resources.

That seems like a good idea to me, but we should wait for others to provide feedback. There may be other plans that would handle that issue I'm not aware of.

danhamill commented:
I think @MikeNeilson had a good point, so I went ahead and updated tests and added resources in hopes this helps expand the functionality of these libraries.

MikeNeilson reviewed (commented):
Overall idea looks good to me. However I have some maintainability concerns.

danhamill commented on ensemble-dss/src/test/resources/hefs_cache/FeatherYuba/1986012612_FeatherYuba_hefs_hourly.csv:
Ha, 4.433 additions and -43 is an alarming PR no doubt.  If we can jump on a call and come up with a way to better store these resources, I am open to it.  

I agree, having the test hefs files in multiple places are excessive. These aren't code.

There is a bit of text duplication across the tests.  Some of it is unavoidable, but one major issue I see is ensemble-dss writes test files to %APPDATA%\Local\Temp\1 and FIRO_TS writes test files to C:\Temp.
